### PR TITLE
Feat add update capability to cluster variable

### LINKED ISSUE #44658: REST API Endpoints for Updating Cluster Variables
Add and document REST API endpoints for updating Cluster Variable values. Ensure endpoints are secure, validate input, and provide reliable error responses for misconfiguration or failed updates.

### PR DESCRIPTION
**Pull request by mathias-vandaele**
This pull request introduces support for updating cluster variables across the stack, including backend services, authorization, validation, database operations, and engine processing. The changes add the ability to update both globally and tenant-scoped cluster variables, ensure proper validation and authorization, and extend the database and engine to handle the new update operation.

**Cluster Variable Update Support**

* Added a new `update` method to `ClusterVariableWriter` and corresponding SQL update statement in `ClusterVariableMapper.xml`, enabling update operations for cluster variables in the database layer. [[1]](diffhunk://#diff-300fd65863f1b323a831d102a152ccce1de48fe4aae2d6dacc7422d2e1b5d109R41-R52) [[2]](diffhunk://#diff-23aa7f8efd8942ac5ad59b089e69b18185286d70d758c88dc265cbf4ca1d7a0eR75-R85)
* Introduced tests for the update operation in `ClusterVariableWriterTest` to ensure correct queuing and truncation behavior.

**API & Validation Enhancements**

* Added support for update requests in the HTTP mapping and validation layers, including new methods in `ClusterVariableMapper` and `ClusterVariableRequestValidator` to process and validate update requests for both global and tenant scopes. [[1]](diffhunk://#diff-b408d3d04bfe5ce82992f1c9f8c5309431f24f8a183cd382865afc2a568d3194R13) [[2]](diffhunk://#diff-b408d3d04bfe5ce82992f1c9f8c5309431f24f8a183cd382865afc2a568d3194R35-R49) [[3]](diffhunk://#diff-5134fc6fe98a113c1a9ea607638f73061f9a280f35f0fc41f2f500361d787c74R13) [[4]](diffhunk://#diff-5134fc6fe98a113c1a9ea607638f73061f9a280f35f0fc41f2f500361d787c74L29-R38) [[5]](diffhunk://#diff-5134fc6fe98a113c1a9ea607638f73061f9a280f35f0fc41f2f500361d787c74R51-R66)
* Refactored validator methods to generalize validation for both create and update operations. [[1]](diffhunk://#diff-8e17ab9af302831d92dd5c66f75ca01a517bda9dc66ca73ce5016d2831f1c44cL23-R23) [[2]](diffhunk://#diff-8e17ab9af302831d92dd5c66f75ca01a517bda9dc66ca73ce5016d2831f1c44cL32-R32)

**Engine & Processing Changes**

* Added a new `ClusterVariableUpdateProcessor` to handle update commands in the engine, including authorization checks and event writing. Registered the processor in `ClusterVariableProcessors`. [[1]](diffhunk://#diff-f8c976e75ff9c83cd84085f7eb4e0cd9e824b75be4ba9620eb70fded44d030c2R32-R36) [[2]](diffhunk://#diff-84cf308aa379e13035eeb1aecbc62dba7f96a0d539f8dbff564fe32ce30497a5R1-R139)
* Implemented a new `ClusterVariableUpdatedApplier` to apply update events to the state.

**Authorization & Permissions**

* Extended cluster variable permissions to include `UPDATE` in both the protocol and the UI permissions configuration. [[1]](diffhunk://#diff-e46384aad9ea42ecf22984a17f15e24def1b79a979ffe8e819bff3f5c271d898L32-R33) [[2]](diffhunk://#diff-1f34912486a365b2caf65e9d0f953e356872cf6dcc53414e93668ac8806f826aR136)

**Service Layer**

* Added service methods to process update requests for both globally and tenant-scoped cluster variables using the new broker request type. [[1]](diffhunk://#diff-1a504cac0da1e50d7ff5bbacf126e66d1abf493f0edae738d800ef791e6f6a25R24) [[2]](diffhunk://#diff-1a504cac0da1e50d7ff5bbacf126e66d1abf493f0edae738d800ef791e6f6a25R96-R113)

These changes collectively enable full support for updating cluster variables, including end-to-end validation, authorization, database persistence, and engine state management.

## Related issues

closes #44658, #44657, #44656

### CODE REVIEW

File: zeebe/gateway-protocol/src/main/proto/v2/cluster-variables.yaml [hunk 1]
```diff
@@ -205,6 +248,55 @@ paths:
                 $ref: 'problem-detail.yaml#/components/schemas/ProblemDetail'
         "500":
           $ref: 'common-responses.yaml#/components/responses/InternalServerError'
+    patch:
+      operationId: updateTenantClusterVariable
+      x-eventually-consistent: false
```
Comment (megglos) [inline zeebe/gateway-protocol/src/main/proto/v2/cluster-variables.yaml]: ❌  this is a full and not partial update of the value, right? Let's use put then
```suggestion
    put:
      operationId: updateTenantClusterVariable
      x-eventually-consistent: false
```

File: zeebe/gateway-protocol/src/main/proto/v2/cluster-variables.yaml [hunk 2]
```diff
@@ -134,6 +134,49 @@ paths:
                 $ref: 'problem-detail.yaml#/components/schemas/ProblemDetail'
         "500":
           $ref: 'common-responses.yaml#/components/responses/InternalServerError'
+    patch:
+      operationId: updateGlobalClusterVariable
+      x-eventually-consistent: false
```
Comment (megglos) [inline zeebe/gateway-protocol/src/main/proto/v2/cluster-variables.yaml]: ❌  this is a full and not partial update of the value, right? Let's use put then
```suggestion
    put:
      operationId: updateGlobalClusterVariable
      x-eventually-consistent: false
```

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/ClusterVariableController.java [hunk 3]
```diff
@@ -89,6 +91,25 @@ public CompletableFuture<ResponseEntity<Object>> deleteTenantClusterVariable(
         .fold(RestErrorMapper::mapProblemToCompletedResponse, this::deleteTenantClusterVariable);
   }
 
+  @CamundaPatchMapping(path = "/global/{name}")
```
Comment (megglos) [inline zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/ClusterVariableController.java]: ❌  in alignment with the spec suggestion
```suggestion
  @CamundaPutMapping(path = "/global/{name}")
```

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/ClusterVariableController.java [hunk 4]
```diff
@@ -89,6 +91,25 @@ public CompletableFuture<ResponseEntity<Object>> deleteTenantClusterVariable(
...
+  public CompletableFuture<ResponseEntity<Object>> updateGlobalClusterVariable(
+      @PathVariable("name") final String name,
+      @RequestBody final UpdateClusterVariableRequest updateClusterVariableRequest) {
+    return clusterVariableMapper
+        .toGlobalClusterVariableUpdateRequest(name, updateClusterVariableRequest)
+        .fold(RestErrorMapper::mapProblemToCompletedResponse, this::updateGlobalClusterVariable);
+  }
+
+  @CamundaPatchMapping(path = "/tenants/{tenantId}/{name}")
```
Comment (megglos) [inline zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/ClusterVariableController.java]: ❌  in alignment with the spec suggestion
```suggestion
  @CamundaPutMapping(path = "/tenants/{tenantId}/{name}")
```

**Review by megglos — CHANGES_REQUESTED** [review]
I focused on the spec changes only.

Please link the relevant issue, I lack transparency on whether you planned to extend the client with a dedicated issue or whether it's missing here.

As the update seems to replace the whole variable value, let's go for the PUT method instead of PATCH.

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ClusterVariableUpdatedApplier.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    this.clusterVariableState = clusterVariableState;
+  }
+
+  @Override
+  public void applyState(final long key, final ClusterVariableRecord clusterVariableRecord) {
+    if (clusterVariableRecord.isTenantScoped()) {
+      clusterVariableState.updateTenantScopedClusterVariable(clusterVariableRecord);
+    } else if (clusterVariableRecord.isGloballyScoped()) {
+      clusterVariableState.updateGloballyScopedClusterVariable(clusterVariableRecord);
+    } // if ever no scope is given, it is better not to do anything for safety
```
Comment (ce-dmelnych) [inline zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ClusterVariableUpdatedApplier.java:30]: 🤔 We can log a `warn` message for such unexpected case

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ClusterVariableUpdatedHandler.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+import io.camunda.util.ClusterVariableUtil;
+import io.camunda.webapps.schema.entities.clustervariable.ClusterVariableEntity;
+import io.camunda.webapps.schema.entities.clustervariable.ClusterVariableScope;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.ClusterVariableIntent;
+import io.camunda.zeebe.protocol.record.value.ClusterVariableRecordValue;
+import java.util.List;
+
+public class ClusterVariableUpdatedHandler
```
Comment (ce-dmelnych) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ClusterVariableUpdatedHandler.java]: 🔧 `ClusterVariableUpdatedHandler` and `ClusterVariableCreatedHandler` handlers are almost identical, WDYT about merging them to single`ClusterVariableCreatedUpdatedHandler`?

As I see this is quite common e.g.:
- [GroupCreatedUpdatedHandler.java](https://github.com/camunda/camunda/blob/203e17f444e89bca76a7205e1cc2547954671a5d/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/GroupCreatedUpdatedHandler.java#L22)
- [UserCreatedUpdatedHandler.java](https://github.com/camunda/camunda/blob/203e17f444e89bca76a7205e1cc2547954671a5d/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/UserCreatedUpdatedHandler.java#L21)
- and several more

File: db/rdbms/src/main/resources/mapper/ClusterVariableMapper.xml [hunk 7]
```diff
@@ -72,6 +72,17 @@
...
   </delete>
 
+  <update id="update" parameterType="io.camunda.db.rdbms.write.domain.ClusterVariableDbModel">
+    UPDATE ${prefix}CLUSTER_VARIABLE
+    SET TYPE = #{type},
+        DOUBLE_VALUE = #{doubleValue, jdbcType=DOUBLE},
+        LONG_VALUE = #{longValue, jdbcType=NUMERIC},
+        VAR_VALUE = #{value},
+        VAR_FULL_VALUE = #{fullValue},
+        IS_PREVIEW = #{isPreview}
```
Comment (ce-dmelnych) [inline db/rdbms/src/main/resources/mapper/ClusterVariableMapper.xml:82]: ❓ Am I correct in understanding that we don't have any scenarios where `TENANT_ID` could be updated for a persisted cluster variable?

File: db/rdbms/src/test/java/io/camunda/db/rdbms/write/service/ClusterVariableWriterTest.java
Comment (ce-dmelnych) [inline db/rdbms/src/test/java/io/camunda/db/rdbms/write/service/ClusterVariableWriterTest.java:1]: ❌ We also need to cover cluster variable update functionality in [ClusterVariableIT.java](https://github.com/camunda/camunda/blob/3f2733cb5748bc02d15605db15bd35f5ef702e1b/qa/acceptance-tests/src/test/java/io/camunda/it/rdbms/db/clustervariables/ClusterVariableIT.java#L38) test.

> coverage for `delete` might also be missing there

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/ClusterVariableController.java
Comment (ce-dmelnych) [inline zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/ClusterVariableController.java:1]: ❌ We are missing coverage for new endpoints in [ClusterVariableControllerTest.java](https://github.com/camunda/camunda/blob/9c4ae20dc23211231750c8786c493c16fae85f69/zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/ClusterVariableControllerTest.java#L37)

File: gateways/gateway-mapping-http/src/main/java/io/camunda/gateway/mapping/http/validator/ClusterVariableRequestValidator.java [hunk 8]
```diff
@@ -47,4 +48,20 @@ public Optional<ProblemDetail> validateTenantClusterVariableRequest(
...
+                name, request.getValue()));
+  }
+
+  public Optional<ProblemDetail> validateTenantClusterVariableUpdateRequest(
+      final String name, final UpdateClusterVariableRequest request, final String tenantId) {
+    return validate(
+        () ->
+            clusterVariableValidator.validateTenantClusterVariableCreateRequest(
+                name, request.getValue(), tenantId));
+  }
```
Comment (ce-dmelnych) [inline gateways/gateway-mapping-http/src/main/java/io/camunda/gateway/mapping/http/validator/ClusterVariableRequestValidator.java:66]: 🔧 The update validation methods are calling `validateGlobalClusterVariableCreateRequest` and `validateTenantClusterVariableCreateRequest`, if the validation is the same for `CreateClusterVariableRequest` and `UpdateClusterVariableRequest` we can adjust the naming for `clusterVariableValidator.validate***` methods.

**Review by ce-dmelnych — CHANGES_REQUESTED** [review]
Hi @mathias-vandaele, thanks for the implementation!

I see this PR covers three related issues (#44656, #44657, #44658) in a single e2E implementation (probably only CamundaClient is out of scope). While this approach has its benefits, for future PRs it would be better to have individual PRs per issue. An iterative approach is safer - it reduces the risk of missing edge cases and has a better effect on test coverage since you're working on a limited scope with more focused attention.

I've left several 🔧suggestions and pointed to places ❌  where additional test coverage should be added.

🔧 Also, the description for this PR is missing 🥲

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ClusterVariableUpdatedApplier.java (hunk 5 above)
Comment (mathias-vandaele) [reply zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ClusterVariableUpdatedApplier.java:30]: Hey @ce-dmelnych I can do that, by adding an else, but this should never happens actually, this is managed in the processor

Comment (mathias-vandaele) [reply zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ClusterVariableUpdatedApplier.java:30]: This was already the case in https://github.com/camunda/camunda/blob/5c7e907dacd24554fd9a3098cc41eb942b0b04cd/zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ClusterVariableCreatedApplier.java#L30

So I believe we can keep it this way 🤔

Comment (ce-dmelnych) [reply zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ClusterVariableUpdatedApplier.java:30]: Yeah, let's keep as is.

File: db/rdbms/src/main/resources/mapper/ClusterVariableMapper.xml (hunk 7 above)
Comment (mathias-vandaele) [reply db/rdbms/src/main/resources/mapper/ClusterVariableMapper.xml:82]: `TENANT_ID` cannot be updated, only value can be, we actually verify the current variable with current tenant exists

File: gateways/gateway-mapping-http/src/main/java/io/camunda/gateway/mapping/http/validator/ClusterVariableRequestValidator.java (hunk 8 above)
Comment (mathias-vandaele) [reply gateways/gateway-mapping-http/src/main/java/io/camunda/gateway/mapping/http/validator/ClusterVariableRequestValidator.java:66]: I replaced them with `validateGlobalClusterVariableRequestWithValue` and `validateTenantClusterVariableRequestWithValue`

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ClusterVariableUpdatedHandler.java (hunk 6 above)
Comment (mathias-vandaele) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ClusterVariableUpdatedHandler.java]: How would that work ? removing the update handler, and just use the Created handler for update and rename it ? I personally prefer to have clear naming, and single handler responsibility, but happy to discuss about it 😀

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/ClusterVariableController.java
Comment (mathias-vandaele) [reply zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/ClusterVariableController.java:1]: True! will do

File: db/rdbms/src/test/java/io/camunda/db/rdbms/write/service/ClusterVariableWriterTest.java
Comment (mathias-vandaele) [reply db/rdbms/src/test/java/io/camunda/db/rdbms/write/service/ClusterVariableWriterTest.java:1]: Those tests are only dedicated to check `ClusterVariableDbReader`, we don't have test for insert here, they are usually associated with the Camunda Client test, WDYT ?

File: db/rdbms/src/main/resources/mapper/ClusterVariableMapper.xml (hunk 7 above)
Comment (ce-dmelnych) [reply db/rdbms/src/main/resources/mapper/ClusterVariableMapper.xml:82]: Got it, thanks for the confirmation 👍🏻

File: db/rdbms/src/test/java/io/camunda/db/rdbms/write/service/ClusterVariableWriterTest.java
Comment (ce-dmelnych) [reply db/rdbms/src/test/java/io/camunda/db/rdbms/write/service/ClusterVariableWriterTest.java:1]: Yeah, we can add those reliable e2e tests in the scope of https://github.com/camunda/camunda/issues/44659.

**Comment by mathias-vandaele** [comment]
@mrm1st3r Could I get a small review from an identity perspective ? https://github.com/camunda/camunda/pull/44735/changes#diff-e46384aad9ea42ecf22984a17f15e24def1b79a979ffe8e819bff3f5c271d898R32

**Comment by mathias-vandaele** [comment]
@camunda/data-layer Could I get small review as there were some changes inside SQL requests ? 💪🏼

**Review by ce-dmelnych — APPROVED** [review]
LGTM 👍🏻 
Thanks for adding more tests.

**Review by mrm1st3r — APPROVED** [review]
Only small changes to identity, LGTM :)

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ClusterVariableCreatedUpdatedHandler.java [hunk 9]
```diff
@@ -15,17 +15,21 @@
...
 import java.util.List;
+import java.util.Set;
 
-public class ClusterVariableCreatedHandler
+public class ClusterVariableCreatedUpdatedHandler
     implements ExportHandler<ClusterVariableEntity, ClusterVariableRecordValue> {
 
-  private static final ClusterVariableIntent SUPPORTED_INTENT = ClusterVariableIntent.CREATED;
+  private static final Set<Intent> SUPPORTED_INTENTS =
+      Set.of(ClusterVariableIntent.CREATED, ClusterVariableIntent.UPDATED);
```
Comment (houssain-barouni) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ClusterVariableCreatedUpdatedHandler.java:27]: ❓ do you know why the CamundaExporter handler does not support the DELETED intent like the RDBMS one?

File: zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/handlers/ClusterVariableCreatedUpdatedHandlerTest.java [hunk 10]
```diff
@@ -71,8 +71,12 @@ void shouldNotHandleRecord(final ClusterVariableIntent intent) {
     assertThat(underTest.handlesRecord(clusterVariableRecordValue)).isFalse();
   }
```
Comment (houssain-barouni) [inline zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/handlers/ClusterVariableCreatedUpdatedHandlerTest.java:72]: ❓ I expected the first test to assert false and the second one to assert true, am I missing something?

**Review by houssain-barouni — APPROVED** [review]
I reviewed the exporters changes + RDBMS update mapper and they LGTM! 👍 
I had some questions though

**Review by megglos — APPROVED** [review]
Spec LGTM! thanks for linking the issues, I know get that client changes come in a follow-up 👍
